scripts/globals/pirates.lua

```diff
@@ -0,0 +1,267 @@
+-----------------------------------
+-- Pirates helpers
+-----------------------------------
+require("scripts/globals/pathfind")
+
+xi = xi or {}
+xi.pirates = xi.pirates or {}
+xi.pirates.status =
+{
+    IDLE       = 0,
+    PREPARING  = 1,
+    ARRIVING   = 2,
+    SPAWNING   = 3,
+    ATTACKING  = 4,
+    DESPAWNING = 5,
+    DEPARTING  = 6,
+    FINISHED   = 7
+}
+
+local function SetZoneMusic(zone, block, track)
+    for _, player in pairs(zone:getPlayers()) do
+        player:ChangeMusic(block, track)
+    end
+end
+
+xi.pirates.init = function(ID)
+    local ship = GetNPCByID(ID.npc.PIRATE_SHIP.id)
+    ship:setAnimPath(ID.npc.PIRATE_SHIP.anim_path)
+    ship:setStatus(xi.status.DISAPPEAR)
+
+    for k, pirate in pairs(ID.npc.PIRATES) do
+        local npc = GetNPCByID(k)
+        npc:setPos(pirate.start_pos)
+        npc:initNpcAi()
+    end
+end
+
+xi.pirates.start = function(ID)
+    local ship = GetNPCByID(ID.npc.PIRATE_SHIP.id)
+    xi.pirates.setStatus(ship, xi.pirates.status.IDLE)
+    xi.pirates.despawnMobs(ID)
+    xi.pirates.despawnNPCs(ID)
+    xi.pirates.reset(ID)
+end
+
+xi.pirates.getStatus = function(ship)
+    return ship:getLocalVar("pirateStatus")
+end
+
+xi.pirates.setStatus = function(ship, status)
+    ship:setLocalVar("pirateStatus", status)
+end
+
+xi.pirates.reset = function(ID)
```

Comment:
So what I would do here, is pass through the zone, then do this:
```
xi.pirates.reset = function(zone)
    local zoneId = zone:getID()
    local mobgroups = zones[zoneId].mob
    local npcs = zones[zoneId].npc.PIRATES
    local ship = GetNPCByID(zones[zoneId].npc.PIRATE_SHIP.id)
    local shipPos =  zones[zoneId].npc.PIRATE_SHIP.start_pos
    ship:setAnimation(0)
    ship:setStatus(xi.status.DISAPPEAR)
    ship:setPos(shipPos.x, shipPos.y, shipPos.z, shipPos.rot)
    for _, group in pairs(mobgroups) do -- I grab the table in the array.
         for _, mob in pairs(group) do -- I grab the IDs from the table.
              if GetMobByID(mob):isSpawned() then
                    DespawnMob(mob)
              end
         end
    end
    for _, npc in pairs(npcs) do
          npc:setStatus(xi.status.DISAPPEAR)
    end
end
```
This ensures we fully cleanup the zone and despawn everything just in case lets say a player only killed the NM and there is still other things up. This also ensures

Reply:
cleanup for the two systems are separate:

Pirates has a cleanup cleaning up pirates and sea_creatures cleans up sea creatures. They should definitely stay separate here.

Reply:
Also putting despawn mob in reset is not what we want. As we want reset and despawning the mobs to be separate for how the ship npc needs to act.

While resetting does need to check and despawn mobs if they are up, we also want to be able to despawn the mobs elsewhere in the script without resetting. Its why they are separate functions.